### PR TITLE
lptim: stm32wb: support of the lptimer

### PR DESCRIPTION
**Pull request by FRASTM**
This patch enables the LP Timer driver for the stm32wb55 soc, to be used in tickless kernel. 
The LPTimer is clocked by the LSE. It provides the following APIs:
z_clock_set_timeout
z_clock_elapsed
z_clock_driver_init
and irq handler

This patch requires the support of the lptimer for l4 serie [#18582]

Signed-off-by: Francois Ramu francois.ramu@st.com

### CODE REVIEW

**Comment by FRASTM** [comment]
This PR requires the "lptim: stm32l4: support of the lptimer" #18582 
The soc stm32wb55 holds the "config STM32_LPTIM_TIMER"
The board set s the CONFIG_STM32_LPTIM_CLOCK_LSE=y
Sample applications could overwrite this in the proj.conf
+

**Comment by FRASTM** [comment]
rebase

**Comment by FRASTM** [comment]
re-submit after update on "lptim: stm32l4: support of the lptimer" #18582

**Comment by FRASTM** [comment]
this PR is following introduction of lptim on l4
lptim: stm32l4: support of the lptimer #18582

File: boards/arm/nucleo_wb55rg/Kconfig.defconfig [hunk 1]
```diff
@@ -10,8 +10,14 @@ if BOARD_NUCLEO_WB55RG
 config BOARD
 	default "nucleo_wb55rg"
 
-config CLOCK_STM32_LSE
```
Comment (erwango) [inline boards/arm/nucleo_wb55rg/Kconfig.defconfig]: Why removing this ?

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_wb55rg/Kconfig.defconfig (hunk 1 above)
Comment (FRASTM) [reply boards/arm/nucleo_wb55rg/Kconfig.defconfig]: restore config CLOCK_STM32_LSE in boards/arm/nucleo_wb55rg/Kconfig.defconfig

**Review by erwango — APPROVED** [review]

**Comment by FRASTM** [comment]
rebase

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Approved by TSC for 2.1 merge

**Pull request merged**